precompiles/common/precompile.go

```diff
@@ -132,7 +135,24 @@ func (p Precompile) RunSetup(
 	// we need to consume the gas that was already used by the EVM
 	ctx.GasMeter().ConsumeGas(initialGas, "creating a new gas meter")
 
-	return ctx, stateDB, s, method, initialGas, args, nil
+	// add a snapshot of the current state before executing the precompile
+	// so that any out-of-gas errors during said execution are reverted correctly
+	if isTransaction(method.Name) {
```

Comment:
So the logic for creating snapshots has been consolidated into a single place here（In the RunSetup method）, clear location (before executing the precompile call) . Errors now consistently return **nil** without carrying snapshots, simplifying the error handling logic.

Reply:
Exactly. Any errors (not just out-of-gas errors) in the precompile call would have prevented the journal entry from being added since that would cause the precompile to exit early. By adding the journal entry before execution of the precompile, we prevent that problem.